app/src/main/kotlin/com/hippo/ehviewer/coil/QRCodeInterceptor.kt

```diff
@@ -0,0 +1,34 @@
+package com.hippo.ehviewer.coil
+
+import coil3.Extras
+import coil3.getExtra
+import coil3.intercept.Interceptor
+import coil3.request.ImageRequest
+import coil3.request.ImageResult
+import coil3.request.SuccessResult
+import com.hippo.ehviewer.image.hasQRCode
+import eu.kanade.tachiyomi.util.lang.withIOContext
+
+private val detectQRCodeKey = Extras.Key(default = false)
+
+fun ImageRequest.Builder.detectQRCode(enable: Boolean) = apply {
+    extras[detectQRCodeKey] = enable
+}
+
+val ImageRequest.detectQRCode: Boolean
+    get() = getExtra(detectQRCodeKey)
+
+object QRCodeInterceptor : Interceptor {
+    override suspend fun intercept(chain: Interceptor.Chain): ImageResult {
+        val result = chain.proceed()
+        if (chain.request.detectQRCode && result is SuccessResult) {
+            val image = result.image
+            if (image is BitmapImageWithExtraInfo) {
+                val hasQRCode = withIOContext { hasQRCode(image.image.bitmap) }
```

Comment:
Why IO?

Reply:
Then `Dispatchers.Default`?

Reply:
Yeah, `detectBorder` needs that as well.
Maybe we just set the global interceptor coroutine context.